### ISSUE TITLE
Undefined variable s:known_types with tagbar interaction

### ISSUE THREAD

**Issue by stah0121** [raven42/devpanel-vim#8]
When opening a file I'm seeing some errors that seem to prevent the tagbar from showing up and when you try to close the file with :q it closes the file but leaves the file explorer open which needs to be :q closed before returning to the terminal.

![image](https://user-images.githubusercontent.com/7742070/210161850-1220d422-96e7-4cd6-bee6-1f07e5cb5af9.png)

I started with a clean install of Ubuntu 22.04 LTS and after some tinkering around I'm confident I have all the necessary packages installed --- afterwards I cloned the vim repository and ran the following commands (note: latest release is v90)

```
126  12/31/2022-23:26:00 cd vim
127  12/31/2022-23:26:07 make distclean
128  12/31/2022-23:26:34 ./configure --prefix=/usr/local --enable-python3interp --enable-perlinterp --enable-gnome-check --enable-gui=auto --enable-gtk2-check --with-x --enable-fontset --enable-gtk2-check
129  12/31/2022-23:27:12 make VIMRUNTIMEDIR=/usr/local/share/vim/vim90
130  12/31/2022-23:30:45 sudo make install
```

The only thing I noticed from the version output is that the perl and the xfontset features didn't get enabled but I'm not sure if that matters for the tagbar? I verified perl is installed (version 5) but am not sure what package might be missing for xfontset

Additional note: I installed the 3 recommended font packages from the dotfiles README but unicode chars weren't working until I installed the "fonts-liberation" package that google-chrome needed ... now the unicode chars seem to be working.

```
wildcat (master) vim$ which vim
/usr/local/bin/vim
wildcat (master) vim$ vi --version
VIM - Vi IMproved 9.0 (2022 Jun 28, compiled Dec 31 2022 23:30:21)
Included patches: 1-1122
Compiled by stah0121@wildcat
Huge version with GTK3 GUI.  Features included (+) or not (-):
+acl               +file_in_path      +mouse_urxvt       -tag_any_white
+arabic            +find_in_path      +mouse_xterm       -tcl
+autocmd           +float             +multi_byte        +termguicolors
+autochdir         +folding           +multi_lang        +terminal
-autoservername    -footer            -mzscheme          +terminfo
+balloon_eval      +fork()            +netbeans_intg     +termresponse
+balloon_eval_term -gettext           +num64             +textobjects
+browse            -hangul_input      +packages          +textprop
++builtin_terms    +iconv             +path_extra        +timers
+byte_offset       +insert_expand     -perl              +title
+channel           +ipv6              +persistent_undo   +toolbar
+cindent           +job               +popupwin          +user_commands
+clientserver      +jumplist          +postscript        +vartabs
+clipboard         +keymap            +printer           +vertsplit
+cmdline_compl     +lambda            +profile           +vim9script
+cmdline_hist      +langmap           -python            +viminfo
+cmdline_info      +libcall           +python3           +virtualedit
+comments          +linebreak         +quickfix          +visual
+conceal           +lispindent        +reltime           +visualextra
+cryptv            +listcmds          +rightleft         +vreplace
+cscope            +localmap          -ruby              +wildignore
+cursorbind        -lua               +scrollbind        +wildmenu
+cursorshape       +menu              +signs             +windows
+dialog_con_gui    +mksession         +smartindent       +writebackup
+diff              +modify_fname      -sodium            +X11
+digraphs          +mouse             -sound             -xfontset
+dnd               +mouseshape        +spell             +xim
-ebcdic            +mouse_dec         +startuptime       -xpm
+emacs_tags        -mouse_gpm         +statusline        +xsmp_interact
+eval              -mouse_jsbterm     -sun_workshop      +xterm_clipboard
+ex_extra          +mouse_netterm     +syntax            -xterm_save
+extra_search      +mouse_sgr         +tag_binary        
-farsi             -mouse_sysmouse    -tag_old_static    
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
  system gvimrc file: "$VIM/gvimrc"
    user gvimrc file: "$HOME/.gvimrc"
2nd user gvimrc file: "~/.vim/gvimrc"
       defaults file: "$VIMRUNTIME/defaults.vim"
    system menu file: "$VIMRUNTIME/menu.vim"
  fall-back for $VIM: "/usr/local/share/vim"
 f-b for $VIMRUNTIME: "/usr/local/share/vim/vim90"
Compilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H -DFEAT_GUI_GTK -pthread -I/usr/include/gtk-3.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0 -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/harfbuzz -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/uuid -I/usr/include/freetype2 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/x86_64-linux-gnu -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -g -O2 -D_REENTRANT -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 
Linking: gcc -L/usr/local/lib -Wl,--as-needed -o vim -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -lharfbuzz -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lSM -lICE -lXt -lX11 -lXdmcp -lSM -lICE -lm -ltinfo -lselinux -L/usr/lib/python3.10/config-3.10-x86_64-linux-gnu -lpython3.10 -lcrypt -ldl -lm -lm 
wildcat (master) vim$
```

**Comment by raven42** [comment]
@stah0121 can you validate if universal-ctags is installed?

```
sudo apt list | grep universal-ctags
```

You should see a listing like this:
```
universal-ctags/jammy,now 5.9.20210829.0-1 amd64 [installed]
```

Specifically the `[installed]` label. If not, then that could be the issue. Tagbar will not fully load if there is no ctags to run and could see an integration problem with the devpanel.

**Comment by stah0121** [comment]
The installed label was missing when I first checked ... and then I deleted the ctags directory and started over running the ctags commands and saw no change.
The last thing I tried was to add 'sudo' to the make install command but that didn't change it either.

This is the session output -- I haven't looked through it very closely ... I will go through and see if there are any errors or problematic warnings later.

[universal-ctags.txt](https://github.com/raven42/devpanel-vim/files/10493911/universal-ctags.txt)

**Comment by raven42** [comment]
If you compile your own, then it will not show installed in the apt list. You can install from the main distro with this if you want instead of compiling your own.
```
sudo apt install universal-ctags
```
Though either way, if you installed manually it should now still be installed. You can verify the output with a `ctags --version`. With ctags installed, does the devpanel still through this error for the tagbar window?

**Comment by stah0121** [comment]
Ah sure, that makes sense.
When I tried `ctags --version` (after compiling my own) it wasn't showing a valid command / output ... so I went ahead and did the `sudo apt install universal-ctags` and that fixed the issue described here with errors showing on file open.

Although now I'm seeing errors when closing a file that are different than this issue ... so I will close this issue and open another one for better tracking.

Thanks for the (continued) help
